src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/adjust_brgemm_copy_b_loop_ports.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/itt.hpp"
+
+#include "adjust_brgemm_copy_b_loop_ports.hpp"
+#include "snippets/utils/utils.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/lowered/expressions/buffer_expression.hpp"
+#include "transformations/snippets/x64/op/brgemm_copy_b.hpp"
+#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+bool pass::AdjustBrgemmCopyBLoopPorts::update_loop_info(const std::shared_ptr<snippets::lowered::UnifiedLoopInfo>& loop_info) {
+    OPENVINO_ASSERT(loop_info, "Invalid loop info pointer");
+    bool modified = false;
+    auto caller = [&](snippets::lowered::LoopPort &loop_port,
+                      snippets::lowered::UnifiedLoopInfo::LoopPortDesc &loop_desc) {
+        const auto& p = *loop_port.expr_port;
+        if (p.get_type() == snippets::lowered::ExpressionPort::Input &&
+            p.get_index() == 1) {
+            const auto& node = p.get_expr()->get_node();
+            if (auto brg = as_type_ptr<BrgemmCPU>(node)) {
+                const auto precision = node->get_input_element_type(1);
+                if (brgemm_utils::with_repacking(brg->get_type()) &&
+                    precision != element::f32 && loop_port.is_incremented) {
+                    // K blocking loop
+                    if (loop_port.dim_idx == 1) {
+                        const auto ptr_incr = loop_desc.ptr_increment;
+                        const auto blocked_shape_ptr_inc = brgemm_utils::repacking::compute_out_leading_dim(ptr_incr, precision);
+                        if (ptr_incr != 0 && ptr_incr != blocked_shape_ptr_inc) {
+                            loop_desc.ptr_increment = blocked_shape_ptr_inc;
+                            OPENVINO_ASSERT(loop_desc.finalization_offset % ptr_incr == 0,
+                                            "Can't rescale finalization offsets");
+                            loop_desc.finalization_offset = loop_desc.ptr_increment *
+                                                             (loop_desc.finalization_offset / ptr_incr);
+                        }
+                    // N blocking loop
```

Comment:
Can we add an explanatory comment here on why do we apply exactly this increment? It is clear for me even without any comments, but only until I forget how the repacked data is stored (the last blocked dim equal to vnni_factor) :)

Reply:
Good idea, added.